activestorage/app/models/active_storage/variation.rb

```diff
@@ -46,7 +46,7 @@ def initialize(transformations)
   end
 
   def default_to(defaults)
-    self.class.new transformations.reverse_merge(defaults)
+    self.class.new defaults.deep_merge(transformations)
```

Comment:
Why do we need to `deep_merge` here?

Reply:
The format for the hash is `{ format: :jpg, saver: { quality: 80, strip: true  }}`. So without the deep merge you lose whatever was in the `saver` hash. One of my tests demonstrates this situation:

```
using_default_transformations(:jpg, { saver: { quality: 80, strip: true } }) do
      blob = create_file_blob(filename: "racecar.jpg")
      variant = blob.variant(resize_to_limit: [100, 100], saver: { quality: 70, interlace: true })
end
```

Without the `deep_merge` the variant will not have `strip: true`.